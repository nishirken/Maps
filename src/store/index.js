export testInitialState from './test-initial-state';
export testServerInitialState from './test-server-initial-state';
export store from './create-store';
export testStore from './test-store';
