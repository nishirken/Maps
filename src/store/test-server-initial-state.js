import { fromJS } from 'immutable';

export default fromJS([
    {
        index: 1,
        coords: {
            lng: 100,
            lat: 200,
            x: 20,
            y: 21,
        },
        name: 'Marker 1',
        objects: [
            {
                index: 0,
                name: 'Object 1',
            },
            {
                index: 1,
                name: 'Object 2',
            },
        ],
    },
    {
        index: 2,
        coords: {
            lng: 200,
            lat: 300,
            x: 10,
            y: 11,
        },
        name: 'Marker 2',
        objects: [
            {
                index: 0,
                name: 'Object 1',
            },
            {
                index: 1,
                name: 'Object 2',
            },
        ],
    },
]);
