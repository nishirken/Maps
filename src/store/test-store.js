import configureStore from 'redux-mock-store';
import initialState from './test-initial-state';

const mockStore = configureStore();
const store = mockStore(initialState.toObject());

export default store;
