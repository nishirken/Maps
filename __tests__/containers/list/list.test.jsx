import { ListContainer } from 'Containers';
import store from 'TestStore';

describe('List container', () => {
    const TestListContainer = shallow(<ListContainer store={store} />);

    it('Rendered', () => {
        expect(TestListContainer).toMatchSnapshot();
    });
});
