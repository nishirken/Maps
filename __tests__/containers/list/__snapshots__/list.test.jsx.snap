// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List container Rendered 1`] = `
<ListContainer
  getCurrentMarker={
    Object {
      "coords": Object {
        "lat": 299,
        "lng": 300,
      },
      "index": 0,
    }
  }
  getMarkerCoords={
    Array [
      Object {
        "coords": Object {
          "lat": 299,
          "lng": 300,
          "x": 200,
          "y": 201,
        },
        "index": 0,
      },
      Object {
        "coords": Object {
          "lat": 300,
          "lng": 301,
          "x": 100,
          "y": 101,
        },
        "index": 1,
      },
    ]
  }
  getMarkerDeleteIndexes={
    Array [
      0,
    ]
  }
  getMarkerName={
    Array [
      Object {
        "index": 0,
        "name": "Marker0",
      },
      Object {
        "index": 1,
        "name": "Marker1",
      },
      Object {
        "index": 1,
        "name": "EditedMarkerName",
      },
    ]
  }
  getMarkerSearchIndexes={
    Array [
      0,
    ]
  }
  getObjectDeleteIndexes={
    Array [
      Object {
        "index": 0,
        "markerIndex": 1,
      },
    ]
  }
  getObjects={
    Array [
      Object {
        "markerIndex": 0,
        "object": Object {
          "index": 0,
          "name": "Object for 0",
        },
      },
      Object {
        "markerIndex": 1,
        "object": Object {
          "index": 0,
          "name": "Object for 1",
        },
      },
      Object {
        "markerIndex": 1,
        "object": Object {
          "index": 1,
          "name": "Object for 1",
        },
      },
    ]
  }
  setCurrentMarker={[Function]}
  setMarkerDeleteIndex={[Function]}
  setMarkerName={[Function]}
  setMarkerSearchIndexes={[Function]}
  setObject={[Function]}
  setObjectDeleteIndex={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
